### ISSUE TITLE
Columns in Windows Explorer details view

### ISSUE THREAD

**Issue by KristinaTaylor** [PTC-ALM/TortoiseSI#36]
For the content available in the file properties page (in #17 ), it should be possible to add this content as additional columns in the details view in windows explorer.

**Comment by KristinaTaylor** [comment]
The question here is: how much work needs to be done, here, after #17 is finished?

**Comment by bleskows** [comment]
Likely to require more research to ascertain if this is actually possible to do. TortoiseSVN had this functionality up until Windows XP, but now ColumnHandlers are not supported from Vista onwards: See [here](http://tortoisesvn.net/docs/release/TortoiseSVN_en/tsvn-dug-wcstatus.html#id663427) under the header "TortoiseSVN Columns In Windows Explorer". Microsoft documentation [here](https://msdn.microsoft.com/en-us/library/bb776831%28v=vs.85%29.aspx) indicating feature no longer supported.

**Comment by bleskows** [comment]
Looking into PropertyHandlers to see if these can provide the functionality

**Comment by bleskows** [comment]
Not looking good. Brace yourselves for a lot of links to look through. As I stated above, IColumnProvider shell extension interface (the column handler) was removed as of Windows Vista (see [Wikipedia entry for features removed in Vista](http://en.wikipedia.org/wiki/List_of_features_removed_in_Windows_Vista#Windows_Explorer)). 

"The IColumnProvider shell extension interface (column handler) has been removed. Shell extensions that display information in Explorer's columns will therefore fail to work. As a result of this API removed, folder sizes cannot be displayed in a column in Windows Explorer Details view. The replacement for column handlers, Property Handlers, cannot be registered for all files."

Microsoft's replacement for Column Handlers are the Property Handlers. Problem - Property Handlers cannot be registered for all files (see this [conversation](https://groups.google.com/forum/#!topic/microsoft.public.platformsdk.shell/55VVuZEMX2g), the user was trying to do almost exactly what we want to do), which if I understand it correctly means that we would have to register a property handler for every single file type we would want to show a custom column for (which would obviously not be feasible). I think the idea of the isColumn attribute for Property Handlers (see [Microsoft documentation](https://msdn.microsoft.com/en-us/library/windows/desktop/bb266532%28v=vs.85%29.aspx)) was to provide functionality for custom columns for custom file types.

**Comment by KristinaTaylor** [comment]
After some reading (thanks!), I think we should probably close this with "wontfix." Several reasons, some of which related to the fact that it doesn't look feasible. More importantly, the more I read, the less it seems like a good idea.

See, for instance, this [article about selfish views of performance](http://blogs.msdn.com/b/oldnewthing/archive/2005/03/11/394249.aspx).

If we load additional data, beyond the status icons, by default, in explorer, we're forcing a bunch of information lookups, probably on some stuff that won't be in the Integrity client's cache, and causing the user to see significant slowdowns when they're just browsing the file system. I don't think the use cases for having this information available at a glance in Windows Explorer are compelling enough to do that.

**Comment by bleskows** [comment]
Agreed. The performance impact would be non-trivial. I know I came across this somewhere (would have to dig it up for a link), but even when the IColumnProvider was supported, Microsoft did not advise tying column values to network-type information (from a UX point of view, shell needs to be fast and should not be limited by the network), as a bad practice. TortoiseSVN/TortoiseGit did not have this problem since their source control methodology is to have the user work on a local repo, that is, where the status is obtained on the client-end.

**Comment by KristinaTaylor** [comment]
I think it's safe to say that we've answered this question.
